New package: CTDirect v0.9.4

JuliaRegistrator opened the pull request:
- Registering package: CTDirect
- Repository: https://github.com/control-toolbox/CTDirect.jl
- Created by: @ocots
- Version: v0.9.4
- Commit: 740c8120b8d0ec9478335053ed59cfbb77fc79df
- Reviewed by: @ocots
- Reference: https://github.com/control-toolbox/CTDirect.jl/issues/156#issuecomment-2212394806
- Description: Direct transcription of an optimal control problem and resolution
<!-- bf0c69308befbd3ccf2cc956ac8a4671113b158fd87a6d4e741b0173dab051a48f7dc43e593ffff4357bc647a3a83598938fcf6bada4b8b5ef40d15ec512ed125438b8e68c88c788169692ea5c2b8ad91c4d292b236ddb3bb892dd3fd0e5bece81ddc3a45d2945d635ac02618353b83974b36ee5421a920a19c8b5cb29f12683b6f3e3a43127c7fef5d02a07d62c6c7a7152ec83dd2bdf3f6e4a14d81f45ce0350e2b7f7b396ce10f47df0463b4e8cdd -->

ocots commented:
[noblock] Dear Admin,

Mainly with @jbcaillau, @PierreMartinon and @joseph-gergaud, we develop an optimal control toolbox :

- [GitHub repo](https://github.com/control-toolbox)
- [Documentation](https://control-toolbox.org/docs/optimalcontrol/stable/)

This toolbox is composed of several packages:

```mermaid
flowchart TD
O(<a href='https://control-toolbox.org/docs/optimalcontrol/stable/'>OptimalControl</a>) --> B(<a href='https://control-toolbox.org/docs/ctbase/stable/'>CTBase</a>)
O --> D(<a href='https://control-toolbox.org/docs/ctdirect/stable/'>CTDirect</a>)
O --> F(<a href='https://control-toolbox.org/docs/ctflows/stable/'>CTFlows</a>)
F --> B
D --> B
```

We intend if accepted to switch from our [local registry](https://github.com/control-toolbox/ct-registry) to the general. We should move `CTBase.jl`, `CTDirect.jl`, `CTFlows.jl` and the main package `OptimalControl.jl`.

Note that we have presented it at the last JuliaCon: see the [presentation](https://control-toolbox.org/docs/optimalcontrol/v0.8/juliacon2023.html).

JuliaRegistrator committed:
New package: CTDirect v0.9.4

UUID: 790bbbee-bee9-49ee-8912-a9de031322d5
Repo: https://github.com/control-toolbox/CTDirect.jl.git
Tree: dee169290e937ba2cdf5a2fdbf2f3d1003ed7f37

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a
